elfo-configurer/src/lib.rs

```diff
@@ -210,30 +210,29 @@ impl Configurer {
         Ok(config_list.into_iter().map(|c| c.group_name).collect())
     }
 
-    async fn request_all<R>(
+    async fn validate_all(
         &self,
         config_list: &[ConfigWithMeta],
-        make_msg: impl Fn(AnyConfig) -> R,
-    ) -> Result<(), Vec<ReloadConfigsError>>
-    where
-        R: Request<Response = Result<(), ConfigRejected>>,
-    {
+    ) -> Result<(), Vec<ReloadConfigsError>> {
         let futures = config_list
             .iter()
             .cloned()
             .map(|item| {
                 let group = item.group_name;
                 let fut = self
                     .ctx
-                    .request_to(item.addr, make_msg(item.config))
+                    .request_to(item.addr, ValidateConfig::new(item.config))
                     .all()
                     .resolve();
 
-                wrap_request_future(fut, group)
+                wrap_long_running_future(
+                    fut,
+                    group,
+                    "group is validing the config suspiciously long",
+                )
             })
             .collect::<Vec<_>>();
 
-        // TODO: use `try_join_all`.
```

Comment:
I am not sure `try_join_all` is the best choice here. According to the [documentation](https://docs.rs/futures/latest/futures/future/fn.try_join_all.html):

```
If any future returns an error then all other futures will be **canceled**...
```

Cancelling a `ValidateConfig` request may result in a situation where only subset of errors are returned to the user. This in turn may lead to the user performing multiple round trips of updating the configs, each time resolving new part of the bigger problem.

What do you think?

Reply:
You're right and it's exactly why `try_join_all` isn't used.